src/Qir/Runtime/public/QirTypes.hpp

```diff
@@ -192,11 +192,12 @@ struct QIR_SHARED_API QirCallable
     void InvokeCaptureCallback(int32_t index, int32_t parameter);
 };
 
-struct QIR_SHARED_API QirRange
+extern "C"
 {
-    int64_t start = 0;
-    int64_t step  = 0;
-    int64_t end   = 0;
-
-    QirRange(int64_t start, int64_t step, int64_t end);
-};
+    struct QirRange
+    {
+        int64_t start;
+        int64_t step;
+        int64_t end;
+    };

```

Comment:
`QIR_SHARED_API` is not needed any more?

Reply:
This one is a little odd, but the short answer is yes, it no longer needs to be `QIR_SHARED_API`. In the QIR spec, `%Range` is just a struct and not a class. Nothing in QIR ever called the constructor function for `QirRange` because it was just created as a struct of three integers directly. The one thing the constructor did was verify that `step` was not zero at construction, but since the constructor isn't called in QIR that doesn't provide any protection. The real protection comes from the `CheckedRange` struct that gets used everyone we actually operate on ranges, specifically this check:
https://github.com/microsoft/qsharp-runtime/blob/d32e2890688a04cb4bca4436deeefba314bc540a/src/Qir/Runtime/lib/QIR/arrays.cpp#L456-L464
Once the constructor is removed and it is made extern C, it can no longer be declared as dllexport, so the `QIR_SHARED_API` had to be removed.